File: java/Example5.java
// import anypackage.in.which.you.put.Exevalator;

/**
 * Function available in expressions.
 */
class MyFunction implements Exevalator.FunctionInterface {

	/**
	 * Invoke the function.
	 * 
	 * @param arguments An array storing values of arguments.
	 * @return The return value of the function.
	 */
	@Override
	public double invoke(double[] arguments) {
		if (arguments.length != 2) {
			throw new Exevalator.Exception("Incorrected number of args");
		}
		return arguments[0] + arguments[1];
	}
}

/**
 * An example to create a function for available in expressions.
 */
public class Example5 {

	/**
	 * The entry point of this example code.
	 * @param args Command-line arguments (unused)
	 */
	public static void main(String[] args) {

		// Create an instance of Exevalator Engine
		Exevalator exevalator = new Exevalator();

		// Connects the function available for using it in expressions
		MyFunction fun = new MyFunction();
		exevalator.connectFunction("fun", fun);

		// Evaluate the value of an expression
		double result = exevalator.eval("fun(1.2, 3.4)");

		// Display the result
		System.out.println("Result: " + result);
	}
}

File: python/example5.py
from exevalator import Exevalator, ExevalatorFunctionInterface, ExevalatorException

class MyFunction(ExevalatorFunctionInterface):
    """Function available in expressions: returns sum of two args."""
    def invoke(self, arguments: list[float]) -> float:
        if len(arguments) != 2:
            raise ExevalatorException("Incorrect number of arguments")
        return arguments[0] + arguments[1]

def main() -> None:
    """
    An example to create a function available in expressions.
    """

    # Create an instance of Exevalator Engine
    ex = Exevalator()

    # Connect the function to use it in expressions
    ex.connect_function("fun", MyFunction())

    # Evaluate the value of an expression
    result = ex.eval("fun(1.2, 3.4)")

    # Display the result
    print(f"result: {result}")   # -> result: 4.6

if __name__ == "__main__":
    main()
